src/coreComponents/constitutive/fluid/chemicalReactions/EquilibriumReactions.cpp

```diff
@@ -0,0 +1,339 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 TotalEnergies
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file EquilibriumReactions.cpp
+ */
+
+#include "EquilibriumReactions.hpp"
+
+#include "functions/FunctionManager.hpp"
+#include "denseLinearAlgebra/interfaces/blaslapack/BlasLapackLA.hpp"
+
+namespace geosx
+{
+
+using namespace stringutilities;
+
+namespace constitutive
+{
+
+namespace chemicalReactions
+{
+
+EquilibriumReactions::EquilibriumReactions( string const & name, integer const numPrimarySpecies, integer const numSecSpecies ):
+  ReactionsBase( name, numPrimarySpecies, numSecSpecies )
+{
+  // Here we should either read the database or the input values.
+
+  // Hardcoding values for now
+
+  
+
+  // Stochiometric Matrix
+  // First index: 0 = OH-, 1 = CO2, 2 = CO3-2, 3 = H2CO3, 4 = CaHCO3+, 5 = CaCO3, 6 = CaSO4, 7 = CaCl+, 8 = CaCl2, 9 = MgSO4, 10 = NaSO4-
+  // Second index: 0 = H+, 1 = HCO3-, 2 = Ca+2, 3 = SO4-2, 4 = Cl-, 5 = Mg+2, 6 = Na+1
```

Comment:
I don't know much about reactive transport, but maybe a third possibility (in addition to reading the database or the input values) would be to keep everything hardcoded in the equilibrium/kinetic reaction classes. 

You would write multiple `EquilibriumReactions` and `KineticReactions` classes with everything hardcoded inside (`numPrimarySpecies`, `numSecSpecies`, stochiometric coefficients), then you would select the right class/template from `ReactiveMultiFluid::createChemicalReactions`.

That could be a way to have a general implementation but still have `matrix`, `solution`, and `rhs` with the right size below. But I don't know enough about reactive transport to know if this is a viable solution.